static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
+    } catch {
+      // TODO: do something?
     }
 
-    fetchGuides();
-  }
+    data?.problems?.forEach?.(problem => {
+      const {id, message, url} = problem;
+      const type = problem.severity === 'critical' ? 'error' : 'warning';
 
-  componentDidUpdate(prevProps) {
-    const {config} = this.props;
-    if (!isEqual(config, prevProps.config)) {
-      this.handleConfigStoreChange(config);
-    }
+      AlertActions.addAlert({id, message, type, url});
+    });
   }
 
-  componentWillUnmount() {
-    OrganizationsStore.load([]);
-    this.unlistener?.();
-  }
+  useEffect(() => {
+    loadOrganizations();
+    checkInternalHealth();
 
-  mainContainerRef = createRef<HTMLDivElement>();
-  unlistener = OrganizationStore.listen(
-    state => this.setState({organization: state.organization}),
-    undefined
-  );
+    // Show system-level alerts
+    config.messages.forEach(msg =>
+      AlertActions.addAlert({message: msg.message, type: msg.level, neverExpire: true})
+    );
 
-  handleConfigStoreChange(config) {
-    const newState = {} as State;
-    if (config.needsUpgrade !== undefined) {
-      newState.needsUpgrade = config.needsUpgrade;
+    // The app is running in deploy preview mode
+    if (DEPLOY_PREVIEW_CONFIG) {
+      displayDeployPreviewAlert();
     }
 
-    if (config.user !== undefined) {
-      newState.user = config.user;
+    // The app is running in local SPA mode
+    if (!DEPLOY_PREVIEW_CONFIG && EXPERIMENTAL_SPA) {
+      displayExperimentalSpaAlert();
     }
 
-    if (Object.keys(newState).length > 0) {
-      this.setState(newState);
+    // Set the user for analytics
+    if (config.user) {
+      HookStore.get('analytics:init-user').map(cb => cb(config.user));
     }
-  }
 
-  @keydown('meta+shift+p', 'meta+k', 'ctrl+shift+p', 'ctrl+k')
-  openCommandPalette(e) {
-    openCommandPalette();
-    e.preventDefault();
-    e.stopPropagation();
-  }
-
-  @keydown('meta+shift+l', 'ctrl+shift+l')
-  toggleDarkMode() {
-    ConfigStore.set('theme', ConfigStore.get('theme') === 'light' ? 'dark' : 'light');
-  }
+    initApiClientErrorHandling();
+    fetchGuides();
 
-  onConfigured = () => this.setState({needsUpgrade: false});
+    // When the app is unloaded clear the organizationst list
+    return () => OrganizationsStore.load([]);
+  }, []);
 
-  // this is somewhat hackish
-  handleNewsletterConsent = () =>
-    this.setState({
-      newsletterConsentPrompt: false,
-    });
+  function clearUpgrade() {
+    ConfigStore.set('needsUpgrade', false);
+  }
 
-  handleGlobalModalClose = () => {
-    if (typeof this.mainContainerRef.current?.focus === 'function') {
-      // Focus the main container to get hotkeys to keep working after modal closes
-      this.mainContainerRef.current.focus();
-    }
-  };
+  function clearNewsletterConsent() {
+    const flags = {...config.user.flags, newsletter_consent_prompt: false};
+    ConfigStore.set('user', {...config.user, flags});
```

Comment:
Update the `ConfigStore` and we don't have to maintain that state here in the component :)